platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/TraceableWiringModel.java

```diff
@@ -14,52 +14,41 @@
-import static com.swirlds.common.wiring.schedulers.builders.TaskSchedulerType.NO_OP;
-import static com.swirlds.common.wiring.schedulers.builders.TaskSchedulerType.SEQUENTIAL_THREAD;
-
-import com.swirlds.common.context.PlatformContext;
-import com.swirlds.common.threading.locks.AutoClosableLock;
-import com.swirlds.common.threading.locks.internal.AutoLock;
-import com.swirlds.common.threading.locks.locked.Locked;
-import com.swirlds.common.wiring.model.ModelEdgeSubstitution;
-import com.swirlds.common.wiring.model.ModelGroup;
-import com.swirlds.common.wiring.model.ModelManualLink;
-import com.swirlds.common.wiring.model.WiringModel;
+
+import com.swirlds.common.wiring.model.diagram.ModelEdgeSubstitution;
+import com.swirlds.common.wiring.model.diagram.ModelGroup;
+import com.swirlds.common.wiring.model.diagram.ModelManualLink;
+import com.swirlds.common.wiring.model.internal.analysis.CycleFinder;
+import com.swirlds.common.wiring.model.internal.analysis.DirectSchedulerChecks;
+import com.swirlds.common.wiring.model.internal.analysis.InputWireChecks;
+import com.swirlds.common.wiring.model.internal.analysis.InputWireDescriptor;
+import com.swirlds.common.wiring.model.internal.analysis.ModelEdge;
+import com.swirlds.common.wiring.model.internal.analysis.ModelVertex;
+import com.swirlds.common.wiring.model.internal.analysis.StandardVertex;
+import com.swirlds.common.wiring.model.internal.analysis.WiringFlowchart;
 import com.swirlds.common.wiring.schedulers.TaskScheduler;
-import com.swirlds.common.wiring.schedulers.builders.TaskSchedulerBuilder;
 import com.swirlds.common.wiring.schedulers.builders.TaskSchedulerType;
-import com.swirlds.common.wiring.schedulers.internal.HeartbeatScheduler;
-import com.swirlds.common.wiring.schedulers.internal.SequentialThreadTaskScheduler;
 import com.swirlds.common.wiring.wires.SolderType;
-import com.swirlds.common.wiring.wires.output.OutputWire;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import edu.umd.cs.findbugs.annotations.Nullable;
-import java.time.Duration;
-import java.time.Instant;
-import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
-import java.util.concurrent.ForkJoinPool;
 
 /**
- * A wiring model is a collection of task schedulers and the wires connecting them. It can be used to analyze the wiring
- * of a system and to generate diagrams.
+ * Common functionality for wiring model implementations. Has methods for registering information about the topology of
+ * wiring that is appropriate for internal use by the framework, but should not be exposed to the end users of the
+ * wiring framework.
  */
-public class StandardWiringModel implements WiringModel {
-
-    /**
-     * The platform context.
-     */
-    private final PlatformContext platformContext;
+public abstract class TraceableWiringModel implements WiringModel {
```

Comment:
In lots of places, you store objects as type `TraceableWiringModel`. Is there a reason you are typing these with the abstract class, instead of the `WiringModel` interface?

Reply:
Yes. Internal things that build wires need access to the methods defined in `TraceableWiringModel`, but I'm intentionally trying to hide these methods from the end user of the wiring model.